src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,931 @@
+      String filePath,
+      int index,
+      String... keys
+    ) {
+      JsonObject obj = files;
+      if (!obj.has(filePath)) {
+        obj.add(filePath, new JsonArray());
+      }
+      JsonArray arr = (JsonArray) obj.get(filePath);
+      while (index >= arr.size()) {
+        arr.add(new JsonObject());
+      }
+      obj = (JsonObject) arr.get(index);
+
+      for (int i = 0; i < keys.length - 1; i++) {
+        String key = keys[i];
+        if (!obj.has(key)) {
+          obj.add(key, new JsonObject());
+        }
+        obj = (JsonObject) obj.get(key);
+      }
+      String lastKey = keys[keys.length - 1];
+      if (!obj.has(lastKey)) {
+        obj.addProperty(lastKey, 0);
+      }
+      return new KeyValuePair<>(lastKey, obj);
+    }
+
+    /**
+     * Adds the passed value to the value found in the files JSON object based on the JSON path created by the
+     * keys. If an object is missing in the JSON path then it's created.
+     * @param value Adds the specified value at the given JSON path.
+     * @param keys The keys in JSON path.
+     */
+    private void safeAddNestedValueInFilesObject(int value, String... keys) {
+      KeyValuePair<String, JsonObject> pair = fetchJsonObjectFromFilesObject(
+        keys
+      );
+      String lastKey = pair.key;
+      JsonObject obj = pair.value;
+      if (!obj.has(lastKey)) {
+        obj.addProperty(lastKey, 0);
+      }
+      JsonPrimitive primitive = (JsonPrimitive) obj.get(lastKey);
+      int number = primitive.getAsInt();
+      obj.addProperty(lastKey, number + value);
+    }
+
+    /**
+     * Adds the passed value to the value found in the files JSON object based on the JSON path created by the
+     * keys. If an object is missing in the JSON path then it's created.
+     * @param value Adds the specified value at the given JSON path.
+     * @param filePath First key in the files object.
+     * @param index Array index of the array the {@code filePath} parameter points to.
+     * @param keys The remaining JSON path keys.
+     */
+    private void safeAddNestedValueInFilesObject(
+      int value,
+      String filePath,
+      int index,
+      String... keys
+    ) {
+      KeyValuePair<String, JsonObject> pair = fetchJsonObjectFromFilesObject(
+        filePath,
+        index,
+        keys
+      );
+      String lastKey = pair.key;
+      JsonObject obj = pair.value;
+      JsonPrimitive primitive = (JsonPrimitive) obj.get(lastKey);
+      int number = primitive.getAsInt();
+      obj.addProperty(lastKey, number + value);
+    }
+
+    /**
+     * Set the passed value to the value found in the files JSON object based on the JSON path created by the
+     * keys. If an object is missing in the JSON path then it's created.
+     * @param value Sets the specified value at the given JSON path.
+     * @param keys The keys in JSON path.
+     */
+    private void safeSetNestedValueInFilesObject(String value, String... keys) {
```

Comment:
You added `safe` here. Do you imply a logic which could be not safe?

Reply:
I called it `safe` because if there are missing objects then they are created for you instead of an exception being thrown.